src/functions/indAffine.jl

```diff
@@ -60,8 +61,12 @@ function (f::IndAffine){R<:Real, T <: RealOrComplex{R}}(x::AbstractArray{T,1})
 end
 
 function prox!{R<:Real, T<:RealOrComplex{R}, M<:DenseArray, V<:AbstractArray{T,1}}(y::V, f::IndAffine{T,M,V,M}, x::V, gamma::R=one(R))
-  res = f.A*x - f.b
-  y[:] = x - f.A'*(f.R\(f.R'\res))
+  A_mul_B!(f.res, f.A, x)
+  f.res .= f.b .- f.res
+  LinAlg.LAPACK.trtrs!('U', 'C', 'N', f.R, f.res);
+  LinAlg.LAPACK.trtrs!('U', 'N', 'N', f.R, f.res);
+  Ac_mul_B!(y, f.A, f.res)
+  y .+= x
```

Comment:
I've actually had significant slowdowns in the code due to lines like this, it seems like the broadcast is not as good as it should. Could you try to run the benchmark with the following code instead:
`using SugarBLAS: @blas!`
line 65:
```julia
@blas! f.res *= - 1.0
@blas! f.res += f.b
```
line 69:
```julia
@blas! y += x
```
At least in my CG code, the following line: `p .= r .+ β.*p` took more time due to all the broadcast than the matrix-vector multiplications... but it was negligible when i changed it to
```julia
@blas! p *= β
@blas! p += r
```